tests/login_test.go

```diff
@@ -182,4 +208,48 @@ var _ = Describe("Login", func() {
 			))
 		})
 	})
+
+	When("Using auth code flow", func() {
+		AfterEach(func() {
+			// reset keyring
+			os.Setenv(properties.KeyringEnvKey, "")
+		})
+
+		It("Fails for an invalid keyring", func() {
+			os.Setenv(properties.KeyringEnvKey, invalidKeyring)
+
+			// Run the command:
+			result := NewCommand().
+				Args(
+					"login",
+					"--use-auth-code",
+				).
+				Run(ctx)
+
+			Expect(result.ExitCode()).ToNot(BeZero())
+			Expect(result.ErrString()).To(ContainSubstring("keyring is invalid"))
+		})
+	})
```

Comment:
Can/should we test the scenario where using the key ring usage does work?

Reply:
yeah, I saw the keyring specific tests, but they all use the offline token, I guess?

Reply:
Ahh I see what you are referencing. Yeah let me see if we can mock these flows and get some coverage for the intersection of the new auth flows and keyrings. The main issue is making sure the tests aren't trying to open a browser during the authentication.

Reply:
got it - so it does add some extra complexity...

FWIW, I did try using both, the use-auth-code login and the key ring config (and I didn't find any issues).

Anyway, I will leave that to you, this is not a blocker to your PR.

Reply:
Added tests to for each keyring against `--use-auth-code`. I'm deferring on `--use-device-code` for now as it will require some significant overhaul to dependency injection against the login command.

Reply:
sounds good, thanks!